### PR TITLE
kyberReserve.js, permissionGroups.js, whitelist.js, volumeImbalanceRecorder.js

### PR DESCRIPTION
**Pull request by Anyhowclick**

### CODE REVIEW

File: test/v4/volumeImbalanceRecorder.js [hunk 1]
```diff
@@ -46,15 +51,15 @@ contract('VolumeImbalanceRecorder', function(accounts) {
         //get token control info
         let controlInfoArr = await imbalanceInst.getTokenControlInfo(token.address);
 
-        assert.equal(controlInfoArr[0].valueOf(), minimalRecordResolution, "unexpected minimal record resolution.");
-        assert.equal(controlInfoArr[1].valueOf(), maxPerBlockImbalance, "unexpected maxPerBlockImbalance.");
-        assert.equal(controlInfoArr[2].valueOf(), maxTotalImbalance, "maxTotalImbalance");
+        Helper.assertEqual(controlInfoArr[0].valueOf(), minimalRecordResolution, "unexpected minimal record resolution.");
+        Helper.assertEqual(controlInfoArr[1].valueOf(), maxPerBlockImbalance, "unexpected maxPerBlockImbalance.");
```
Comment (ilanDoron) [inline test/v4/volumeImbalanceRecorder.js]: please don't use valueOf if not necessary.
check if necessary...

File: test/v4/kyberReserve.js [hunk 2]
```diff
@@ -207,196 +209,196 @@ contract('KyberReserve', function(accounts) {
...
+            totalExpectedTwei = totalExpectedTwei.add(expectedTweiAmount);
             reserveTokenBalance[tokenInd] = reserveTokenBalance[tokenInd].sub(expectedTweiAmount);
 
             let reportedBalance = await token.balanceOf(reserveInst.address);
             await reserveInst.trade(ethAddress, amountWei, tokenAdd[tokenInd], user1, buyRate, true, {from : network, value:amountWei});
-            let recordImbal = expectedTweiAmount.div(minimalRecordResolution).floor().mul(minimalRecordResolution);
+            let recordImbal = expectedTweiAmount.div(minimalRecordResolution).mul(minimalRecordResolution);
             reserveTokenImbalance[tokenInd] = reserveTokenImbalance[tokenInd].add(recordImbal); //imbalance represents how many missing tokens
-            totalWei += (1 * amountWei);
+            totalWei = totalWei.add(amountWei);
```
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply test/v4/kyberReserve.js]: Not sure why it doesn't work.... Hmmm....

File: test/v4/kyberReserve.js [hunk 3]
```diff
@@ -407,13 +409,13 @@ contract('KyberReserve', function(accounts) {
         //verify sell rate
         let sellRate = await reserveInst.getConversionRate(tokenAdd[tokenInd], ethAddress, amountTwei, currentBlock);
 
-        let expectedRate = (new BigNumber(baseSellRate[tokenInd]));
+        let expectedRate = baseSellRate[tokenInd];
         let extraBps = getExtraBpsForSellQuantity(amountTwei);
-        expectedRate = addBps(expectedRate, extraBps);
-        expectedRate.floor();
+        expectedRate = Helper.addBps(expectedRate, extraBps);
+        expectedRate;
```
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply test/v4/kyberReserve.js]: oh.

File: test/v4/kyberReserve.js [hunk 4]
```diff
@@ -499,22 +500,22 @@ contract('KyberReserve', function(accounts) {
...
-            let expectedRate = (new BigNumber(baseSellRate[tokenInd])).floor();
+            let expectedRate = baseSellRate[tokenInd];
             let extraBps = getExtraBpsForSellQuantity(amountTwei);
-            expectedRate = addBps(expectedRate, extraBps);
+            expectedRate = Helper.addBps(expectedRate, extraBps);
             extraBps = getExtraBpsForImbalanceSellQuantity((reserveTokenImbalance[tokenInd].sub(amountTwei)));
-            expectedRate = addBps(expectedRate, extraBps);
-            expectedRate = expectedRate.floor();
+            expectedRate = Helper.addBps(expectedRate, extraBps);
+            expectedRate = expectedRate;
```
Comment (ilanDoron) [inline test/v4/kyberReserve.js]: this line should be removed...
will merge and do it.

**Pull request merged**